docs/Conformance-Guide-draft.md

```diff
@@ -0,0 +1,103 @@
+
+   Such registration events will be logged by Zowe Explorer and can be used to verify to successful implementation of the requirement in the Zowe Explorer logs.
+
+1. (Best Practice) Matching Zowe CLI Plugin: **Provide a Zowe CLI Plugin for the data provider's profile type that implements the core capabilities requires for the new protocol that users can then also use to interact with the protocol outside of the Zowe Explorer extension using Zowe CLI commands.**
+
+   A Zowe CLI Plugin is a regular NPM package that then can be completely imported as a dependency into a VS Code extension that represents a data provider for Zowe Explorer. The zFTP Zowe CLI Plugin and Zowe Explorer extension for zFTP are an example for such a pairing. This requirement can be verified by testing the CLI plugin and seeing it being used in the package.json file of the VS Code extension.
+
+1. (Required) Data provider API implementation: **The Zowe Explorer extensibility API provides interfaces for the required operations to be implemented for MVS, JESS and USS. An extender must either fully implement and register at least one of these three interfaces or alternatively throw JavaScript exceptions that provide meaningful error messages to the end-user in the `Error.message` property of the exception that Zowe Explorer will display in a dialog.**
+
+   "Not yet implemented by the XYZ data provider" could be an example. Registering a new data provider API instance will be logged by Zowe Explorer and be be verified in the Zowe Explorer log files.
+
+1. (Best practice) API test suite implementation: **If the extension implements a Zowe Explorer API data provider interface, it should implement a test suite that calls each of the implemented API methods.**
+
+   Test run logs can be provided to demonstrate the fulfillment of the requirement.
+
+## Menu extension conformance criteria (3)
+
+1. (Required) Menu Names: **If the extension is adding new commands and context menu entries to the Zowe Explorer tree view nodes, it adheres to the Zowe Explorer-provided contextual string format.**
+
+   **TBD**: Define rules and verification criteria.
+
+   _For team discussion_: We currently only have documented for how to add context menus, but not how the commands that implement the menu can actually access the node on which the click occurred to determine all the information needed (e.g. the data set member name and the profile of the node). We need to implement and provide an API for our tree browsers to properly support this and prevent users for just hacking the tree views themselves. Until we have such an API perhaps this extensibility kind needs to be postponed.
```

Comment:
Colin's [test mule repo](https://github.com/Colin-Stone/zowe-explorer-test-mule/blob/master/src/extension.ts#L51) demonstrates how it's currently possible for extenders to access the information for the node that was clicked on. Here is a snippet of the information I saw his extender accessing, which includes data set name and profile information:

<img width="481" alt="Screen Shot 2021-05-05 at 8 11 38 AM" src="https://user-images.githubusercontent.com/45975633/117139589-6cecb680-ad7a-11eb-9f79-76aca148d9e3.png">

Perhaps with an API, we could enforce better control over how much and what kind of node information the extenders can access.

However, as extenders seem to already be able to access the information and hook into our context menus, I was thinking maybe we want to provide some guidelines on how we want extenders to define groupings for their context menu items (i.e. in relation to Zowe Explorer's existing context menu groupings).